### DISCUSSION TITLE
TechEmpower Framework Benchmarks

### DISCUSSION THREAD

**Discussion by dukefirehawk** [dart-backend/angel#25] [Announcements]
Angel3 is officially on TechEmpower as of 15 Dec 2021. The preliminary results can be found at: 

https://www.techempower.com/benchmarks/#section=test&runid=0666746d-6a7b-472c-ba1d-cd9f7a70bb43&hw=ph&test=db&a=2

Not too bad considering, there is almost no optimization for handling large number of concurrent access. No view template caching, no database connections pooling, no proper utilization of isolates etc. That said a lot of room for improvement.

Immediately following this, the goal is
1. Add the cached queries test with angel3_cache and angel3_redis to complete all 7 tests
2. Add a new set of test with angel3_mongo instead of angel3_orm + postgresql

The results will provide good insight into what to change in order to improve Angel3 framework performance. May not be the best but at least above average is achievable.

**Comment by denisano** [comment]
Very intresting and encouraging results! The higher position the higher famouse. Looking forward the optimizations!
